src/ensembl/src/content/app/entity-viewer/gene-view/components/default-transcripts-list/transcripts-list-item-info/TranscriptsListItemInfo.test.tsx

```diff
@@ -3,21 +3,24 @@ import { mount } from 'enzyme';
 
 import TranscriptsListItemInfo from './TranscriptsListItemInfo';
 
+import { createGene } from 'tests/fixtures/entity-viewer/gene';
 import { createTranscript } from 'tests/fixtures/entity-viewer/transcript';
 
 describe('<TranscriptsListItemInfo /', () => {
   let wrapper: any;
+  const transcript = createTranscript();
+  const gene = createGene({ transcripts: [transcript] });
+  const props = {
+    gene,
+    transcript
+  };
 
   beforeEach(() => {
-    const props = {
-      transcript: createTranscript()
-    };
-
     wrapper = mount(<TranscriptsListItemInfo {...props} />);
   });
 
   it('displays amino acid length when transcript has CDS', () => {
-    expect(wrapper.find('#amino-acid-length')).toHaveLength(1);
+    expect(wrapper.find('.topMiddle strong')).toHaveLength(1);
```

Comment:
Should the `strong` tag have some className specified? 
I am asking because if we change the content to be something else in future instead of `amino acid length`, this test will still pass.

Reply:
There are several layers to this question:
1) What to do when we do not have a sufficiently unique identifier
2) Are CSS selectors good enough at all
3) Is a test that just looks at the presence of a particular element on the page useful at all.

Your question "what if we change the content to be something else" refers to all three of them. Even if an element has a class name associated with it, when we change the content of this element to be something else, without changing the class of the container element, the test would still pass.

I agree that `.topMiddle strong` is rather non-specific. What I'd like to avoid (going back to sub-question 1) is having classes that are not associated with any styling. What we **definitely** should avoid is adding actual ids to non-unique elements, because ids should be unique per page. This is why I changed the selector in this test.

Some people solve this problem by adding the `data-test-id` attribute to respective elements, and some even clean up these otherwise useless attributes from the production build with something like [this plugin](https://www.npmjs.com/package/babel-plugin-jsx-remove-data-test-id). I don't know if we want to move in this direction.

Reply:
I agree that there is no point in adding a classname when they are not associated with any styling.
I don't have a strong opinion to which way to go for but the only thing I can think of is, in future, if a new developer wants to investigate a wrong value displayed here, it will be more easier to search where `#amino-acid-length` is used rather than `topMiddle` or `strong`. It will be straightforward for those who clearly knows the directory structure but for others, I guess not. This is something I do for the current site to find where certain values are used.

Reply:
I've updated the test to be more rigorous. I kept the selector the same, but I am now checking the actual content of the element showing amino acids count.

Incidentally, when I updated the test to actually check the content of the element, I found that the code in the component had an error — it was returning an incorrect number of amino acids if a transcript has a single exon. It's a perfect demonstration of how checking simply for the presence of a DOM element is not a particularly useful test in many circumstances 😄 Given that it's temporary code anyway, I added a quick-and-dirty fix to the component.

The conversation in this thread is very important, but I suggest we discuss our attitude towards selectors in testing separately and maybe create a separate PR with appropriate changes. I am leaning towards admitting that in some circumstances we might need to resort to `data-test-id`s.